src/main/java/teammates/logic/api/GateKeeper.java

```diff
@@ -346,14 +346,17 @@ private boolean isStudent() {
         return StudentsLogic.inst().isStudentInAnyCourse(user.getNickname());
     }
 
-    public void verifyAccessibleForCurrentUserAsInstructor(AccountAttributes account, String courseId, String section) {
+    public void verifyAccessibleForCurrentUserAsInstructorOrTeamMember(AccountAttributes account, String courseId,
+            String section, String email) {
         InstructorAttributes instructor = InstructorsLogic.inst().getInstructorForGoogleId(courseId, account.googleId);
-
-        if (instructor == null) {
-            throw new UnauthorizedAccessException("User is not instructor of the course that student belongs to");
-        } else if (!instructor.isAllowedForPrivilege(section,
-                                                     Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)) {
+        StudentAttributes student = StudentsLogic.inst().getStudentForCourseIdAndGoogleId(courseId, account.googleId);
+        if (instructor == null && student == null) {
+            throw new UnauthorizedAccessException("User is not in the course that student belongs to");
+        } else if (instructor != null && !instructor.isAllowedForPrivilege(section,
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_VIEW_STUDENT_IN_SECTIONS)
+                || student != null && !StudentsLogic.inst().isStudentsInSameTeam(courseId, email, student.email)) {
             throw new UnauthorizedAccessException("User does not have enough privileges to view the photo");
+
```

Comment:
Let's do it this way:
```java
if (instructor != null) {
    if (...) {
        throw new UnauthorizedAccessException(...);
    }
    return;
}

StudentAttributes student = ...;

if (student != null) {
    if (...) {
        throw new UnauthorizedAccessException(...);
    }
    return;
}

throw new UnauthorizedAccessException(...);
```
This way we can see the hierarchy more clearly, suppresses the need to query for extra student entity if the instructor entity is non-null, and also allows for different exception messages for instructor and student (they should not be the same).